### PR TITLE
feat(types): Add organization ID parsing for DSNs

### LINKED ISSUE #1159: Parse organization ID from DSN
Add a `Dsn::org_id()` helper so the SDK can derive its organization ID from SaaS DSNs as required by the [strict trace continuation spec](<https://develop.sentry.dev/sdk/foundations/trace-propagation/#the-sdks-organization-id>). This is the first implementation task in the strict trace continuation stack tracked by getsentry/sentry-rust#1016.

The helper should parse SaaS-style hosts whose first DNS label is `o<digits>` and return `None` for self-hosted, Relay, malformed, or overflow cases. Add a `CHANGELOG.md` entry because this is a public API addition.

<details><summary>Implementation notes</summary>

Target files:

* `sentry-types/src/dsn.rs`
* `sentry-types/src/protocol/v7.rs` for the existing `OrganizationId` type
* `CHANGELOG.md`

Implementation:

* Add `Dsn::org_id(&self) -> Option<OrganizationId>`.
* Parse only the first host label.
* Accept labels matching `o<digits>`.
* Return `None` for labels such as `oabc`, `123`, empty labels, self-hosted hosts, Relay hosts, and integer overflow.

Examples:

* `o123.ingest.sentry.io` -> `Some(123.into())`
* `o123.sentry.io` -> `None`
* `sentry.io` -> `None`
* `relay.example.com` -> `None`
* `localhost` -> `None`
* `oabc.ingest.sentry.io` -> `None`

Tests:

* Add unit tests in `sentry-types/src/dsn.rs` for valid SaaS hosts, malformed first labels, self-hosted hosts, and overflow.
* Validate with `cargo test -p sentry-types dsn`.

</details>

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
Expose `Dsn::org_id` so callers can derive the SDK organization ID needed for strict trace continuation. Parse only Sentry SaaS ingest DSN hosts, including regional hosts, and return `None` for self-hosted, Relay, malformed, and overflow cases.

Closes [#1159](https://github.com/getsentry/sentry-rust/issues/1159)
Closes [RUST-233](https://linear.app/getsentry/issue/RUST-233)

### CODE REVIEW

**Review by giortzisg — APPROVED** [review]

File: sentry-types/src/dsn.rs [hunk 1]
```diff
@@ -214,6 +228,35 @@ impl FromStr for Dsn {
...
+/// Given a Sentry host, extracts the org ID as a `&str`.
+///
+/// Returns [`None`] if this is not a Sentry host.
+///
+/// Sentry hosts must look like one of the following:
+///     - `o{orgid}.ingest.{region}.sentry.io`
+///     - `o{orgid}.ingest.sentry.io`
+fn extract_org_id(host: &str) -> Option<&str> {
+    // First split on the dots, up to 6 segments.
+    let mut host_iter = host.splitn(6, '.');
```
Comment (Litarnus) [inline sentry-types/src/dsn.rs:240]: Just curious, are you splitting it into 6 segments to make sure the last one is always `None`?

File: sentry-types/src/dsn.rs [hunk 2]
```diff
@@ -317,6 +360,88 @@ mod test {
...
+
+    #[cfg(feature = "protocol")]
+    #[test]
+    fn test_dsn_org_id_rejects_overflow() {
+        assert_org_id(
+            "https://username@o18446744073709551616.ingest.sentry.io/42",
+            None,
+        );
+    }
+
```
Comment (Litarnus) [inline sentry-types/src/dsn.rs:444]: You might want to add a test for a hostname with a 6th part that is not `None`, e.g. `https://username@o123.ingest.de.sentry.io.com/42`

**Review by Litarnus — APPROVED** [review]

**Review by giortzisg — APPROVED** [review]

File: sentry-types/src/dsn.rs (hunk 1 above)
Comment (szokeasaurusrex) [reply sentry-types/src/dsn.rs:240]: Yes, precisely. Otherwise, you could imagine a host like `o123.ingest.us.sentry.io.this.is.not.valid` being accepted
